primitives/header-chain/src/lib.rs

```diff
@@ -32,8 +38,25 @@ pub mod justification;
 pub trait Parameter: Codec + EncodeLike + Clone + Eq + Debug {}
 impl<T> Parameter for T where T: Codec + EncodeLike + Clone + Eq + Debug {}
 
-/// A base trait for pallets which want to keep track of a full set of headers from a bridged chain.
-pub trait BaseHeaderChain {
+/// A GRANDPA Authority List and ID.
+#[derive(Default, Encode, Decode, RuntimeDebug, PartialEq, Clone)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct AuthoritySet {
+	/// List of GRANDPA authorities for the current round.
+	pub authorities: AuthorityList,
+	/// Monotonic identifier of the current GRANDPA authority set.
+	pub set_id: SetId,
+}
+
+impl AuthoritySet {
+	/// Create a new GRANDPA Authority Set.
+	pub fn new(authorities: AuthorityList, set_id: SetId) -> Self {
+		Self { authorities, set_id }
+	}
+}
+
+/// base trait for verifying transaction inclusion proofs.
+pub trait TransactionVerifier {
```

Comment:
Are you sure it is a good name? :) I'd imagine that `TransactionVerifier` is verifying transactions, not transaction inclusion proofs

Reply:
It seemed like an okay name - but I'm open to changing it to something better. Maybe `TransactionProofVerifier`?